packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputShadowNode.java

```diff
@@ -263,6 +264,10 @@ public void setPadding(int spacingType, float padding) {
    * {@code EditText} this class uses to determine the expected size of the view.
    */
   protected EditText createInternalEditText() {
-    return new EditText(getThemedContext());
+    // By setting a style which has a background drawable, this EditText will have a different
+    // background drawable instance from that on the UI Thread, which maybe has a default background
+    // drawable instance.
+    // Otherwise, DrawableContainer is not a thread safe class, and it caused the npe in #29452.
+    return new EditText(getThemedContext(), null, R.attr.rnTextInputStyle);
```

Comment:
Is the idea here that `getThemedContext()` returns resources shared with the UI thread, but this EditText we use for measurement on Paper runs in a background thread?

Reply:
yes

Reply:
This is the stacktrace of issue points to ReactEditText https://github.com/facebook/react-native/issues/17530
My interpretation was that the NPE triggers with the change of state of a ReactEditText (AppCompatEditText) https://chromium.googlesource.com/android_tools/+/7200281446186c7192cb02f54dc2b38e02d705e5/sdk/extras/android/support/v7/appcompat/res/drawable/abc_edit_text_material.xml#25

```java
java.lang.NullPointerException:
tempt to invoke virtual method 'android.graphics.drawable.Drawable android.graphics.drawable.Drawable$ConstantState.newDrawable(android.content.res.Resources)' on a null object reference at
android.graphics.drawable.DrawableContainer$DrawableContainerState.getChild(DrawableContainer.java:888) at
android.graphics.drawable.DrawableContainer.selectDrawable(DrawableContainer.java:466) at
android.graphics.drawable.StateListDrawable.onStateChange(StateListDrawable.java:104) at
android.graphics.drawable.Drawable.setState(Drawable.java:735) at
android.graphics.drawable.DrawableWrapper.onStateChange(DrawableWrapper.java:331) at
android.graphics.drawable.Drawable.setState(Drawable.java:735) at
android.graphics.drawable.LayerDrawable.onStateChange(LayerDrawable.java:1488) at
android.graphics.drawable.Drawable.setState(Drawable.java:735) at
android.view.View.drawableStateChanged(View.java:18002) at
android.widget.TextView.drawableStateChanged(TextView.java:4097) at
android.view.View.refreshDrawableState(View.java:18071) at
android.view.View.setPressed(View.java:8543) at
android.view.View.setPressed(View.java:8521) at
android.view.View.onTouchEvent(View.java:11218) at
android.widget.TextView.onTouchEvent(TextView.java:8467) at
com.facebook.react.views.textinput.ReactEditText.onTouchEvent(ReactEditText.java:150)
```